pkg/app/piped/controller/scheduler.go

```diff
@@ -635,6 +643,21 @@ func (s *scheduler) reportDeploymentCompleted(ctx context.Context, status model.
 	return err
 }
 
+func (s *scheduler) getMentionedAccounts(ctx context.Context, status model.DeploymentStatus) ([]string, error) {
+	ds, err := s.targetDSP.GetReadOnly(ctx, ioutil.Discard)
+	if err != nil {
+		err = fmt.Errorf("Failed to prepare running deploy source data (%v)", err)
+		return nil, err
+	}
+
+	for _, v := range ds.GenericDeploymentConfig.DeploymentNotification.Mentions {
+		if event := "EVENT_" + v.Event; event == model.NotificationEventType_name[int32(status.Number())] {
```

Comment:
This logic is a bit complex. If anyone has ideas, please let me know.

Reply:
well, the key of `NotificationEventType_name` and `DeploymentStatus` isn't matched at all.

```go
	DeploymentStatus_DEPLOYMENT_PENDING      DeploymentStatus = 0
	DeploymentStatus_DEPLOYMENT_PLANNED      DeploymentStatus = 1
	DeploymentStatus_DEPLOYMENT_RUNNING      DeploymentStatus = 2
	DeploymentStatus_DEPLOYMENT_ROLLING_BACK DeploymentStatus = 3
	DeploymentStatus_DEPLOYMENT_SUCCESS      DeploymentStatus = 4
	DeploymentStatus_DEPLOYMENT_FAILURE      DeploymentStatus = 5
	DeploymentStatus_DEPLOYMENT_CANCELLED    DeploymentStatus = 6
```

```go
	NotificationEventType_name = map[int32]string{
		0:   "EVENT_DEPLOYMENT_TRIGGERED",
		1:   "EVENT_DEPLOYMENT_PLANNED",
		2:   "EVENT_DEPLOYMENT_APPROVED",
		3:   "EVENT_DEPLOYMENT_ROLLING_BACK",
		4:   "EVENT_DEPLOYMENT_SUCCEEDED",
		5:   "EVENT_DEPLOYMENT_FAILED",
		6:   "EVENT_DEPLOYMENT_CANCELLED",
		7:   "EVENT_DEPLOYMENT_WAIT_APPROVAL",
		100: "EVENT_APPLICATION_SYNCED",
		101: "EVENT_APPLICATION_OUT_OF_SYNC",
		200: "EVENT_APPLICATION_HEALTHY",
		300: "EVENT_PIPED_STARTED",
		301: "EVENT_PIPED_STOPPED",
	}
```

Reply:
You are right, but what we need to match are three statuses(`DEPLOYMENT_SUCCESS`, `DEPLOYMENT_FAILURE`, `DEPLOYMENT_CANCELLED`. Because they matched, I implemented such as above.

Reply:
They are lol
But actually they're matched just by chance. `NotificationEventType` and `DeploymentStatus` are totally different concept each other. It's quite hard to maintain long-term.

The number of events you want to handle is few, right? So how about matching manually here, like:

```go
	switch status {
	case model.DeploymentStatus_DEPLOYMENT_SUCCESS:
            expected = model.NotificationEventType_EVENT_DEPLOYMENT_SUCCEEDED
	case model.DeploymentStatus_DEPLOYMENT_FAILURE:
		...
		
	}
```

Reply:
That sounds nice. Thank you for giving me good advice!

Reply:
96548e6